### PR TITLE
Add OkunaDesktop

### PR DESCRIPTION
**Pull request by 999eagle**

### CODE REVIEW

**Comment by LeandroStanger** [comment]
bot, build moe._999eagle.OkunaDesktop

File: moe._999eagle.OkunaDesktop.json [hunk 1]
```diff
@@ -0,0 +1,82 @@
+{
+	"app-id": "moe._999eagle.OkunaDesktop",
+	"runtime": "org.freedesktop.Platform",
+	"runtime-version": "18.08",
```
Comment (LeandroStanger) [inline moe._999eagle.OkunaDesktop.json]: Update runtime to 19.08

File: moe._999eagle.OkunaDesktop.json [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+					"type": "file",
+					"path": "build.sh"
+				},
+				{
+					"type": "file",
+					"path": "okuna-desktop.sh"
+				},
+				{
+					"type": "file",
+					"path": "moe._999eagle.OkunaDesktop.appdata.xml"
```
Comment (LeandroStanger) [inline moe._999eagle.OkunaDesktop.json]: New appdata files should use metainfo instead. https://www.freedesktop.org/software/appstream/docs/chap-Quickstart.html

File: build.sh [hunk 3]
```diff
@@ -0,0 +1,34 @@
+#!/usr/bin/env bash
```
Comment (barthalion) [inline build.sh]: Ideally, these should be put into `build-commands` instead of being separate script.

File: build.sh [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+echo "copying files"
+
+# copy main okuna files
+mv okuna-desktop /app/opt/
+
+# set up entry point
+mv okuna-desktop.sh /app/bin/
+rm /app/opt/okuna-desktop/okuna-desktop.sh
+
+# set up desktop entry
```
Comment (barthalion) [inline build.sh]: As you are the upstream developer, is there any reason why these changes can't happen in code repository instead? Especially the icon part. `Exec=` should be fixed automatically by flatpak-builder.

File: moe._999eagle.OkunaDesktop.json [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+				"/lib/pkgconfig"
+			],
+			"config-opts": [
+				"--enable-shared",
+				"--enable-pic",
+				"--disable-lavf"
+			],
+			"sources": [
+				{
+					"type": "archive",
```
Comment (barthalion) [inline moe._999eagle.OkunaDesktop.json]: Does it contain binaries for i386, x86_64, arm and aarch64? Otherwise you need to create `flathub.json` file to specify for which arch application should be built: https://github.com/flathub/flathub/wiki/App-Maintenance#limiting-the-set-of-architectures-to-build-on

File: moe._999eagle.OkunaDesktop.appdata.xml [hunk 6]
```diff
@@ -0,0 +1,34 @@
+<?xml version="1.0" encoding="utf-8" ?>
```
Comment (barthalion) [inline moe._999eagle.OkunaDesktop.appdata.xml]: Appdata file is also usually something that would be preferred to be maintained in upstream repo; also other distributions use it.

Comment (999eagle) [reply moe._999eagle.OkunaDesktop.appdata.xml]: The appdata is maintained in the upstream repo, but while I'm still working on Flatpak, it's not yet on master. It will be though.

File: moe._999eagle.OkunaDesktop.json (hunk 5 above)
Comment (999eagle) [reply moe._999eagle.OkunaDesktop.json]: No, I only support x86_64 due to libraries I use. I'll add that, thanks.

File: build.sh (hunk 4 above)
Comment (999eagle) [reply build.sh]: I thought it'd be more useful to do it like this, but I can also rely on `rename-icon=` in the manifest. Will change that.

File: build.sh (hunk 3 above)
Comment (999eagle) [reply build.sh]: Okay, will do.

File: moe._999eagle.OkunaDesktop.json (hunk 2 above)
Comment (999eagle) [reply moe._999eagle.OkunaDesktop.json]: Okay, will change that.

File: moe._999eagle.OkunaDesktop.appdata.xml [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+<!-- Copyright 2019 Sophie Tauchert -->
+<component type="desktop-application">
+	<id>moe._999eagle.OkunaDesktop</id>
```
Comment (nedrichards) [inline moe._999eagle.OkunaDesktop.appdata.xml]: why the underscore here? the domain is just https://999eagle.moe/ right?

Comment (999eagle) [reply moe._999eagle.OkunaDesktop.appdata.xml]: Flatpak builder didn't accept a domain component starting with a number. I though this was the best way to do it.

Comment (nedrichards) [reply moe._999eagle.OkunaDesktop.appdata.xml]: oh interesting, I wonder if that's an inherited dbus requirement or something else. I'll have a quick dig. But yeah, that's a legit answer.

Comment (barthalion) [reply moe._999eagle.OkunaDesktop.appdata.xml]: Yeah, that's dbus thing.
